### ISSUE TITLE
Move data from training to live

### ISSUE THREAD

**Issue by DavidRoy** [BiologicalRecordsCentre/ABLE#596]
Request to transfer some counts that were made on training mode to 'live'

240215	- A Rocha Cruzinha I
240222	- A Rocha Cruzinha II
243008	- Mata dos Sete Montes
240301	- Parque Ambiental de Santa Margarida
256572	- ICNF PNA Serra do Louro
272901	- ICNF RNES Gambia

**Comment by CrisSevilleja** [comment]
Hello again. How is this task going?
I have several users that would like to change, training mode surveys into live surveys to appear properly on the website and database. 

- User ID 332002, around 605 records are in training mode mentioned on #639 
- User ID 297313, 158 surveys (15min counts) in training mode to change into live (Romania)
- User ID 326171, 13 transect surveys in training mode. (needed for Slovenia coordinator)

**Comment by JimBacon** [comment]
1. Records for user 326171 updated with the following query.

```sql
with to_change as (
	select p.id p_id, p.training p_training, s.id s_id, s.training, o.id o_id, o.training
	from cache_samples_functional s
	join cache_occurrences_functional o
		on o.sample_id = s.id
	join cache_samples_functional p
		on p.id = s.parent_sample_id
	where s.created_by_id = 326171
	and s.survey_id = 562
	order by s.parent_sample_id
),
update_o as (
	update occurrences o
	set training = false, updated_on = now(), updated_by_id = 2
	from to_change
	where to_change.o_id = o.id 	
),
update_s as (
	update samples s
	set training = false, updated_on = now(), updated_by_id = 2
	from to_change
	where to_change.s_id = s.id 	
),
update_s as (
	update samples p
	set training = false, updated_on = now(), updated_by_id = 2
	from to_change
	where to_change.p_id = p.id 	
),
update_p_cache as (
	insert into work_queue(task, entity, record_id, params, cost_estimate, priority, created_on)
	select distinct on (p_id) 'task_cache_builder_update', 'sample', p_id, null, 100, 2, now()
	from to_change
	order by p_id
),
update_s_cache as (
	insert into work_queue(task, entity, record_id, params, cost_estimate, priority, created_on)
	select distinct on (s_id) 'task_cache_builder_update', 'sample', s_id, null, 100, 2, now()
	from to_change
	order by s_id
),
update_o_cache as (
	insert into work_queue(task, entity, record_id, params, cost_estimate, priority, created_on)
	select 'task_cache_builder_update', 'occurrence', o_id, null, 100, 2, now()
	from to_change
	order by o_id
)
select * from work_queue
```

**Comment by JimBacon** [comment]
2. Records for user 297313 updated as above but with the following substitution:

```sql
with to_change as (
	select p.id p_id, p.training p_training, s.id s_id, s.training, o.id o_id, o.training
	from cache_samples_functional s
	join cache_occurrences_functional o
		on o.sample_id = s.id
	join cache_samples_functional p
		on p.id = s.parent_sample_id
	where s.created_by_id = 297313
	and s.survey_id = 565
	and s.training = true
	order by s.parent_sample_id
)
```

**Comment by JimBacon** [comment]
3. Records for user 332002 updated as above but with the following filter.

```sql
	where s.created_by_id = 332002
	and s.training = true
```

**Comment by JimBacon** [comment]
4. 900 occurrence records for location 240215 were updated as above but with the following filter:

```sql
	where p.location_id = 240215
	and s.survey_id = 562
	and o.training = true
```

**Comment by JimBacon** [comment]
5. And similar to above, updated
    1. 568 occurrence records for location 240222,
    2. 406 for location 243008 
    3. 147 for location 240301 
    4. 27 for location 256572 
    5.  65 for location 272901 

This completes all the actions requested in this issue.

**Comment by zdfric** [comment]
Hello, I face the same problems with users 381903 and 326549, the change from training to non-training mode did not help to see their samples. Can you please update their data so it is visible?

**Comment by DavidRoy** [comment]
@JimBacon can you update the training records to non-training for users 381903 and 326549

**Comment by zdfric** [comment]
@DavidRoy & @JimBacon thank you very much! I am sorry to disturb you, but I found that these recorders (348015 and 346731) also have their data hidden as they set the "training" by error.

**Comment by JimBacon** [comment]
Github is not currently showing my comment where I noted that user 381903 had no records to update but 30 visits by user 326549 were updated.

- For user 348015, 20 transect counts and 23 15-minute counts were updated.
- For user 346731, 24 transect counts were updated.

**Comment by CrisSevilleja** [comment]
Thank Jim for moving these training records to live. I have other two users to change data into live:

- user 303008, transect surveys for several years (Austria for me to keep the track)
- user 253383, transect surveys for several years (Italy)

thank you in advance.

**Comment by CrisSevilleja** [comment]
@JimBacon 
another user changing training data to live plus the previous ones on pending:
- [x] user ID 240329, Transect ID 222006 several surveys (Portugal)
- [x] user 303008, transect surveys for several years (Austria for me to keep the track)
- [x] user 253383, transect surveys for several years (Italy)

**Comment by JimBacon** [comment]
I have updated the records as requested.

**Comment by CrisSevilleja** [comment]
Hello @JimBacon I have two users with training data to move to live:

- user 419921 with 15 surveys of 15 min counts in training mode this year
- user 332002 with 5 15min counts from 2023. For this user, you already changed his training records into live, but it is remaining these 4 surveys with no butterflies. Maybe zero count surveys are not included in the script.

Thank you in advance. I reopen this issue to keep ¡track of the users better in one issue. Cheers

**Comment by zdfric** [comment]
Hello, @JimBacon, 
Several Czech users are still having some records as training data, please, can you set the users & their data to non-training?
transects:
Blažková, Zuzana (381903) 
Adam, Otomar (411260)

15-min counts:
Faltynek Fric, Zdenek (246969)
Hanč, Zdeněk (348015)

The trouble with the first user is also repeated from the previous years
Thank you very much!

**Comment by JimBacon** [comment]
Hi @zdfric 

The training mode setting is in the mobile app. Each user needs to disable training mode on their phone.
See https://www.youtube.com/watch?v=8RrPsdrdLbk&list=PLi6Kp1S3lc5JjheyG298bLFOpYOpfO4yP&index=1

Once this is done and they are no longer submitting training records then I will update the database.
Regards,
Jim Bacon.

**Comment by CrisSevilleja** [comment]
Hello @zdfric and @JimBacon 

I checked the users on the website, and nobody is in training mode on the website. Checking the training data, there are just a few counts/visits done with the training mode on, and it looks like it was accidental at the beginning of using eBMS. 

Therefore, if @JimBacon can change the training data into live would be ideal. And also @zdfric, if you can check with the users, they are not in training mode in the app. 

Thank you in advance!

**Comment by zdfric** [comment]
@CrisSevilleja & @JimBacon,
Thanks a lot! None of the users are in training mode; I have already checked (one of the users is me!). 
Please, who in the development team can solve the issue with phones? One of our users purchased a Redmi M2004J19C phone in the summer with Android 15. Since the Butterfly Count feature does not work on this device, it always results in a blank screen. He tried several times to uninstall and reinstall Butterfly Count, and the issue is still there.
Regards,

Zdenek

**Comment by DavidRoy** [comment]
@zdfric please raise a new issue for the app problem, choosing the option to report an app issue. We'll then assign to Karolis to investigate

**Comment by JimBacon** [comment]
I have made the following changes.

- User 381903 has no records in the database so there was nothing to change. They were in training mode on the website so I turned this off.
- User 411260 had one transect visit (29281959) to Horní Stropnice-Hojná Voda on 23/4/2025 in training mode which has been changed to non-training.
- User 246969 had three 15-minute samples from 2020 (7871394, 844106, 844107) in training mode which have been changed to non-training.
- User 348015 had two 15-minute samples from 2023 (21475911, 22311575) in training mode which have been changed to non-training. They had no observations attached.
  They also have 17 transect visits with transect sections in training mode but with no observations. These were not changed.

In total, only 20 species observations were affected and none of them were very recent. This does not feel consistent with the problem I felt you were reporting. Does this solve your issues @zdfric ?

**Comment by zdfric** [comment]
Dear @JimBacon ,
Thanks a lot! I see I made an error regarding the first user, the correct user ID is 346731; please, can you check the data of this user and shift them to live data? BTW this is strange that there were still these old records in the training mode.



<img width="960" height="189" alt="Image" src="https://github.com/user-attachments/assets/bd6bfcf6-3efd-441e-be5f-1514bb6c81e1" />


> I have made the following changes.
> 
> * User 381903 has no records in the database so there was nothing to change. They were in training mode on the website so I turned this off.
> * User 411260 had one transect visit (29281959) to Horní Stropnice-Hojná Voda on 23/4/2025 in training mode which has been changed to non-training.
> * User 246969 had three 15-minute samples from 2020 (7871394, 844106, 844107) in training mode which have been changed to non-training.
> * User 348015 had two 15-minute samples from 2023 (21475911, 22311575) in training mode which have been changed to non-training. They had no observations attached.
>   They also have 17 transect visits with transect sections in training mode but with no observations. These were not changed.
> 
> In total, only 20 species observations were affected and none of them were very recent. This does not feel consistent with the problem I felt you were reporting. Does this solve your issues [@zdfric](https://github.com/zdfric) ?

**Comment by JimBacon** [comment]
@zdfric  
- User 346731 had 4 transect visits in training mode (27878496, 27878499, 27878502, 27878552). There were also some sections in training mode even though the parent transect visit was not in training mode. All of these have been set to non-training. There were no species observations attached to any of these samples.
- User 411260 had a second transect visit in training mode (29281899) which my original query did not detect as it has no observations attached. This has now been set to non-training.

**Comment by zdfric** [comment]
@JimBacon , thank you very much!!!!

**Comment by CrisSevilleja** [comment]
thank you Jim

**Comment by JimBacon** [comment]
@CrisSevilleja, from your request of 27/5/2025

- User 419921: Fifteen 15-minute counts changed from training to live.
- User 332002: Five 15-minute counts with no occurrences changed from training to live.

**Comment by CrisSevilleja** [comment]
great @JimBacon thank you!
